contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```
@@ -0,0 +1,415 @@
+ * provided then reflection is used to determine this mapping. As of now only
+ * primitive types are supported.
+ *
+ * @displayName Avro To Pojo
+ * @category Converter
+ * @tags avro
+ * @since 3.3.0
+ */
+@InterfaceStability.Evolving
+public class AvroToPojo extends BaseOperator
+{
+
+  private Class<?> cls;
+
+  private static final String FIELD_SEPARATOR = ":";
+  private static final String RECORD_SEPARATOR = ",";
+
+  private String genericRecordToPOJOFieldsMapping = null;
+
+  private List<FieldInfo> fieldInfos;
+
+  private List<ActiveFieldInfo> columnFieldSetters = null;
+
+  @AutoMetric
+  int recordCnt = 0;
+
+  @AutoMetric
+  int errorCnt = 0;
+
+  @AutoMetric
+  int fieldErrorCnt = 0;
+
+  /**
+   * Retruns a string representing mapping from generic record to POJO fields
+   */
+  public String getGenericRecordToPOJOFieldsMapping()
+  {
+    return genericRecordToPOJOFieldsMapping;
+  }
+
+  /**
+   * Comma separated list mapping a field in Avro schema to POJO field eg :
+   * orderId:orderId:INTEGER
+   */
+  public void setGenericRecordToPOJOFieldsMapping(String genericRecordToPOJOFieldsMapping)
+  {
+    this.genericRecordToPOJOFieldsMapping = genericRecordToPOJOFieldsMapping;
+  }
+
+  @InputPortFieldAnnotation(optional = false)
+  public final transient DefaultInputPort<GenericRecord> data = new DefaultInputPort<GenericRecord>()
+  {
+    @Override
+    public void process(GenericRecord tuple)
+    {
+      processTuple(tuple);
+    }
+  };
+
+  /**
+   * Converts given Generic Record and to a POJO and emits it
+   */
+  protected void processTuple(GenericRecord tuple)
+  {
+    try {
+      Object obj = getPOJOFromGenericRecord(tuple, getCls());
+
+      if (obj != null) {
+        output.emit(obj);
+        recordCnt++;
+      }
+
+    } catch (InstantiationException | IllegalAccessException e) {
+      LOG.error("Could not initialize object of class - " + getClass().getName());
```

Please do not hide the exception, atleast log the exception.